src/main/java/com/exadel/discountwebapp/vendor/service/VendorService.java

```diff
@@ -55,16 +55,19 @@ public VendorResponseVO create(VendorRequestVO request) {
 
     @Transactional(propagation = Propagation.REQUIRED)
     public VendorResponseVO update(Long id, VendorRequestVO request) {
-        vendorValidator.validate(request);
-        Vendor vendor = getVendorById(id);
-        vendorMapper.update(vendor, request);
-        vendorRepository.save(vendor);
-        return vendorMapper.toVO(vendor);
+        var vendor = getVendorById(id);
+        if (!vendor.getEmail().equals(request.getEmail())) {
+            vendorValidator.validate(request);
+        }
+        vendorMapper.update(request, vendor);
+        var updatedVendor = vendorRepository.save(vendor);
+        return vendorMapper.toVO(updatedVendor);
     }
 
     @Transactional(propagation = Propagation.REQUIRED)
     public void deleteById(Long id) {
-        vendorRepository.deleteById(id);
+        var vendor = getVendorById(id);
+        vendorRepository.deleteById(vendor.getId());
```

Comment:
we don't need to getVendorBy Id first

we can simply try to deleteById

If there is no vendor, and we will try to delete it, the operation will be successfull because the result is what we want. There will be no vendor with such id in DB

Reply:
fixed